### DISCUSSION TITLE
Clarify Project

### DISCUSSION THREAD

**Discussion by pkavajin** [hofstadter-io/hof#73] [General]
```
"The hof tool builds on Cuelang and leverages single-source of truth designs and data models, code generation, flexible resource and runtime systems, sharable workspaces, contexts, dev setups, and much much more."
```

No offense, but what is this project about?
Neither description nor website makes clear what this is about :/

**Comment by verdverm** [comment]
yeah, I know, it's been a persistent point of feedback and I've still been more in the code trying to get a first "api" together before really promoting. Things have been evolving enough (settling soon) that I have been putting of docs / site. That will be changing in the next couple of weeks.

In brief (or perhaps ramble):

- started out as a code generator, the idea that I can write my conceptual design for something software implemented down once, and then generate as much code as possible. Then use diff3 for allowing custom code and generated code to live in the same files, rather granularly. I do try to minimize the actual code mixing as much as possible, and typically just have a simple enough function call at that crossing point.
- I devops a lot, so there is huge influence from ideas there. Largely that we should declare our intent and have systems worry about how to make that a reality. Also kubernetes CRDs and label systems. Much of devops is configuring software that is ready to run, and in many ways we configure components or modules for software.
- Cuelang entered when my yaml based config (designs / dsl ...) UX/DX broke down and I went looking for something better. Cue is definitely special. It's theory is more sound than all of the other experiments going on around configuration in devops. It also comes from the best heritage and experience from my survey of the landscape. I am having similar experiences with Cue that I did with Go, largely that it was so much easier to express what I wanted with less text / keypresses and more reliability.
- Since then, the tool has become something of a swiss army knife for a number of ideas, which I think would provide a better overall DX if all woven together... so hof is becoming a monotool of sorts. It will cover those 80% use cases with good defaults and let you get detailed when you need to.
- `hof` is headed towards becoming the open source, developer oriented project. We also have `_saas` as the open source business side, largely focused on helping B2B operations. This is also built to fill our own needs because we have a product called `studios` which is (in cue) `#studios: hof & _saas & { people, marketing, sales, support, success... }`

Documentation is coming soon, `hof`'s built in help system is, and always will be, the most current source. Technically this is under `design/`, but I am planning to generate documentation from the tool (and designs too probably).
